wallet/src/main/scala/org/bitcoins/wallet/internal/FundTransactionHandling.scala

```diff
@@ -147,6 +143,9 @@ trait FundTransactionHandling extends WalletLogger { self: Wallet =>
 
         }
       }
+      _ <-
+        if (markAsReserved) markUTXOsAsReserved(addrInfosWithUtxo.map(_._1))
```

Comment:
Even here things can fail later on in the `for` expression. We probably want to do something like 

```scala

 val resultF = for { ... } yield { ... }

resultF.recoverWith { _ => 
  //unreserve utxos since we failed to sign
  //any other cleanup things?
}

resultF
```

Reply:
Would like to be able to do that but we don't have access to which utxos we selected outside of the future.